recipe_scrapers/irishcentral.py

```diff
@@ -0,0 +1,113 @@
+import json, re
+from ._exceptions import FieldNotProvidedByWebsiteException
+from ._utils import normalize_string
+
+class IrishCentral(AbstractScraper):
+    @classmethod
+    def host(cls):
+        return "irishcentral.com"
+
+    def author(self):
+        script_tag = self.soup.find('script', type='application/ld+json')
+
+        if script_tag:
+            script_content = script_tag.string
+            cleaned_script_content = re.sub(r'[\x00-\x1f\x7f]', '', script_content)
+            try:
+                ld_json = json.loads(cleaned_script_content)
+                author = ld_json.get('author', {}).get('name', 'Author not found')
+                return author
+            except json.JSONDecodeError:
+                raise FieldNotProvidedByWebsiteException(return_value=None)
+        else:
+            raise FieldNotProvidedByWebsiteException(return_value=None)
+
+    def description(self):
+        description = self.soup.find('meta', {'property': 'og:description'})['content']
+        return description if description else FieldNotProvidedByWebsiteException(return_value=None)
+
+    def image(self):
+        image = self.soup.find('meta', {'property': 'og:image'})['content']
+        return image if image else FieldNotProvidedByWebsiteException(return_value=None)
+
+    def ingredients(self):
+        # Check if the ingredients are in a <p> structure (https://www.irishcentral.com/culture/food-drink/apple-jameson-tart-recipe)
+        ingredients_label = self.soup.find("p", text=re.compile(r"Ingredients:"))
+
+        if ingredients_label:
+            ingredients_list = []
+            ingredients_paragraphs = ingredients_label.find_next_siblings("p")
+
+            for paragraph in ingredients_paragraphs:
+                text = normalize_string(paragraph.get_text())
+
+                if not text or text.lower() in ["most read", "popular"]:
+                    continue
+
+                if not text.startswith("-") and not any(char.isdigit() for char in text):
+                    break
+
+                ingredients_list.append(text.lstrip("-").strip())
+
+            if ingredients_list:
+                return ingredients_list
+
+            # Check if the ingredients are in a <ul> structure (https://www.irishcentral.com/culture/food-drink/shepherds-pie-recipe)
+            ingredients_list = ingredients_label.find_next("ul")
+            if ingredients_list:
+                ingredients_list = [
+                    normalize_string(li.get_text())
+                    for li in ingredients_list.find_all("li")
+                    if li.get_text(strip=True) and li.get_text(strip=True).lower() not in ["most read", "popular"]
+                ]
+                if ingredients_list:
+                    return ingredients_list
+
+        return FieldNotProvidedByWebsiteException(return_value=None)
+
+    def instructions(self):
+        instructions_label = self.soup.find("p", text=re.compile(r"Method:"))
```

Comment:
```suggestion
        instructions_label = self.soup.find("p", string=re.compile(r"Method"))
        instructions_list = []
```
`DeprecationWarning: The 'text' argument to find()-type methods is deprecated. Use 'string' instead.` & moving initialization of instructions_list to the top of the method